etna/analysis/outliers/rolling_statistics.py

```diff
@@ -275,3 +276,118 @@ def get_anomalies_iqr(
     )
 
     return _outliers_per_segment(df=df, func=detection_func, index_only=index_only)
+
+
+@sliding_window_decorator
+def mad_method(
+    series: pd.Series,
+    indices: np.ndarray,
+    mad_scale: float = 1.5,
+    period: Optional[int] = None,
+    trend: bool = False,
+    seasonality: bool = False,
+    stl_params: Optional[Dict[str, Any]] = None,
+) -> np.ndarray:
+    """
+    Estimate anomalies using MAD statistics.
+    Parameters
+    ----------
+    series:
+        original series for the estimation.
+    indices:
+        which observations use for the estimation.
+    mad_scale:
+        scaling parameter of the estimated interval.
+    period:
+        periodicity of the sequence for STL.
+    trend:
+        whether to remove trend from the series.
+    seasonality:
+        whether to remove seasonality from the series
+    stl_params:
+        other parameters for STL. See https://www.statsmodels.org/devel/generated/statsmodels.tsa.seasonal.STL.html
+    Returns
+    -------
+    :
+        binary mask for each observation, indicating if it was estimated as an anomaly.
+    """
+    if mad_scale <= 0:
+        raise ValueError("Scaling parameter must be positive!")
+
+    window = series[indices]
+
+    if trend or seasonality:
+        if stl_params is None:
+            stl_params = {}
+
+        window = _stl_decompose(series=window, period=period, trend=trend, seasonality=seasonality, **stl_params)
+
+    mad = median_abs_deviation(window)
+    median = np.median(window)
+
+    anom_mask = (window > median + mad_scale * mad) | (window < median - mad_scale * mad)
```

Comment:
```suggestion
    anom_mask = np.abs(window - median) >  mad * mad_scale
```
Let's speed up things a bit here.